src/client.ts

```diff
@@ -1339,17 +1343,35 @@ export class StreamChat<
     }
 
     /** update the client.state with any changes to users */
-    if (event.type === 'user.presence.changed' || event.type === 'user.updated') {
-      if (event.user?.id === this.userID) {
-        this.user = this.user && { ...this.user, ...event.user };
-        /** Updating only available properties in _user object. */
-        Object.keys(event.user).forEach((key) => {
-          if (this._user && key in this._user) {
-            /** @ts-expect-error */
-            this._user[key] = event.user[key];
-          }
-        });
+    if (
+      (event.type === 'user.presence.changed' || event.type === 'user.updated') &&
+      event.user.id === this.userID
+    ) {
+      // Remove deleted properties from user objects.
+      for (const key in this.user) {
+        if (key in event.user || isOwnUserBaseProperty(key)) {
+          continue;
+        }
+
+        if (this.user?.[key]) {
+          delete this.user[key];
```

Comment:
hey maybe create a shallow copy of user and delete properties on it instead of deleting this.user[key] directly? this way we can only set this.user in the end of the function so we will only mutate it once instead of manually deleting the keys during the function execution.
this is useful to avoid half-updated states in case this function fails in the middle of its execution.

Reply:
makes sense :)